Client/Frontend/Browser/Tab Management/TabManagerStore.swift

```diff
@@ -244,12 +248,85 @@ class TabManagerStoreImplementation: TabManagerStore, FeatureFlaggable, Loggable
         _ = imageStore?.clearExcluding(savedUUIDs)
         return savedTabs.isEmpty ? nil : savedTabs
     }
+
+    private func write(tabStateData: Data?) {
+        guard let data = tabStateData, let path = tabsStateArchivePath() else { return }
+        do {
+            try data.write(to: path, options: [])
+        } catch {
+            // Failure could happen when restoring
+            self.browserLog.debug("PreserveTabs write failed with bytes count: \(data.count)")
+        }
+    }
+
+    // MARK: - Deprecated
+    // To remove once migration is completed, see FXIOS-4912 && FXIOS-4913
+
+    var deprecatedTabsToMigrate: [SavedTab] {
+        guard let tabData = deprecatedTabDataRetriever.getTabData() else { return [SavedTab]() }
+
+        // In case tabs aren't migrated to Codable yet
+        // We'll be able to remove this when adoption rate to v106 and greater is high enough
+        let deprecatedUnarchiver = NSKeyedUnarchiver(forReadingWith: tabData)
+        deprecatedUnarchiver.setClass(SavedTab.self, forClassName: "Client.SavedTab")
+        deprecatedUnarchiver.setClass(SessionData.self, forClassName: "Client.SessionData")
+        deprecatedUnarchiver.decodingFailurePolicy = .setErrorAndReturn
+        guard let migratedTabs = deprecatedUnarchiver.decodeObject(forKey: tabsKey) as? [SavedTab] else {
+            let error = String(describing: deprecatedUnarchiver.error)
+            return savedTabError(description: "Deprecated unarchiver could not decode Saved tab with: \(error)")
+        }
+        return migratedTabs
+    }
```

Comment:
nit: computed properties are great however in this case, since we are touching keyed unarchiver and this is already called from the computed property (tabs) it would be nice to make it into a method just for cleanup.